packages/fs-detectors/test/unit.local-file-system-detector.test.ts

```diff
@@ -5,7 +5,7 @@ import { LocalFileSystemDetector, DetectorFilesystem } from '../src';
 
 const tmpdir = path.join(os.tmpdir(), 'local-file-system-test');
 
-const dirs = ['', 'a', 'a/b']; // root, single-nested, double-nested
+const dirs = ['', 'a', `a${path.sep}b`]; // root, single-nested, double-nested
```

Comment:
I wonder if this was intentionally `/` on all systems?

Should we use `path.posix.join()` instead of `path.join()`?

Reply:
tbh I have no idea. `DetectorFilesystem` is not very well defined IMO. Maybe @Ethan-Arrowood @agadzik @chloetedder can clarify.

Reply:
In a perfect world I would vote to redesign the API for `DetectorFilesystem#readdir()` such that it would _only_ return the filenames, and not have the directory context with it. That would match POSIX behavior. If we wanted to do that then we would need to coordinate with the API side as well since this interface is used there as well.

Reply:
I'm pretty sure that we don't use any directory info from readdir for our detection utils, just file info